src/main/java/com/sourcegraph/javagraph/Project.java

```diff
@@ -2,12 +2,24 @@
 
 import java.nio.file.Path;
 import java.util.List;
-import java.util.Set;
 
 public interface Project {
```

Comment:
Can you add a concise docstring for this interface and docstring for each of its methods describing its functionality, purpose, and how to use it?